pkg/virt-operator/kubevirt_test.go

```diff
@@ -2796,6 +2808,131 @@ var _ = Describe("KubeVirt Operator", func() {
+
+				apiDeployment := getDefaultVirtApiDeployment(NAMESPACE, customConfig)
+				controllerDeployment := getDefaultVirtControllerDeployment(NAMESPACE, customConfig)
+				handlerDaemonset := getDefaultVirtHandlerDaemonSet(NAMESPACE, customConfig)
+				// omitempty ignores the field's zero value resulting in the json patch test op breaking
+				apiDeployment.ObjectMeta.Generation = 123
+				controllerDeployment.ObjectMeta.Generation = 123
+				handlerDaemonset.ObjectMeta.Generation = 123
+
+				kvTestData.addDeployment(apiDeployment, kv)
+				kvTestData.addDeployment(controllerDeployment, kv)
+				kvTestData.addDaemonset(handlerDaemonset, kv)
+				kvTestData.addPodsAndPodDisruptionBudgets(customConfig, kv)
+				kvTestData.makeDeploymentsReady(kv)
+				kvTestData.makeHandlerReady()
+
+				deploy := &appsv1.Deployment{}
+				kvTestData.deploymentPatchReactionFunc = func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+					a := action.(testing.PatchActionImpl)
+					patch, err := jsonpatch.DecodePatch(a.Patch)
+					Expect(err).ToNot(HaveOccurred())
+
+					o, exists, err := kvTestData.controller.stores.DeploymentCache.GetByKey(fmt.Sprintf("%s/%s", NAMESPACE, a.Name))
+					Expect(err).ToNot(HaveOccurred())
+					Expect(exists).To(BeTrue())
+					existing := o.(*appsv1.Deployment)
+					existingDeploymentBytes, err := json.Marshal(existing)
+					Expect(err).ToNot(HaveOccurred())
+
+					targetDeploymentBytes, err := patch.Apply(existingDeploymentBytes)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(json.Unmarshal(targetDeploymentBytes, deploy)).To(Succeed())
+
+					return true, deploy, nil
+				}
+
+				ds := &appsv1.DaemonSet{}
+				kvTestData.daemonSetPatchReactionFunc = func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+					a := action.(testing.PatchActionImpl)
+					patch, err := jsonpatch.DecodePatch(a.Patch)
+					Expect(err).ToNot(HaveOccurred())
+
+					o, exists, err := kvTestData.controller.stores.DaemonSetCache.GetByKey(fmt.Sprintf("%s/%s", NAMESPACE, a.Name))
+					Expect(err).ToNot(HaveOccurred())
+					Expect(exists).To(BeTrue())
+					existing := o.(*appsv1.DaemonSet)
+					existingDeploymentBytes, err := json.Marshal(existing)
+					Expect(err).ToNot(HaveOccurred())
+
+					targetDeploymentBytes, err := patch.Apply(existingDeploymentBytes)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(json.Unmarshal(targetDeploymentBytes, ds)).To(Succeed())
+
+					return true, ds, nil
+				}
+
+				kvTestData.shouldExpectPatchesAndUpdates(kv)
+				kvTestData.shouldExpectKubeVirtUpdateStatus(1)
+				kvTestData.shouldExpectCreations()
+
+				kvTestData.controller.Execute()
+
+				Expect(kvtesting.FilterActions(&kvTestData.kubeClient.Fake, "patch", "deployments")).To(HaveLen(2))
+				Expect(kvtesting.FilterActions(&kvTestData.kubeClient.Fake, "patch", "daemonsets")).To(HaveLen(1))
+
+				for _, meta := range []metav1.Object{deploy, ds, &deploy.Spec.Template, &ds.Spec.Template} {
```

Comment:
In fact you are only testing one of the deployments here which is either virt-api or virt-controller no?

Reply:
right, I think so